### DISCUSSION TITLE
Creating Cert...

### DISCUSSION THREAD

**Discussion by simpsonetti** [Ylianst/MeshCentral#5723] [Q&A]
Hi there, 
i want now to use MC in a production environment. Therefor i want to use certificates. We have a CA. 
So i'm asking me, how is the "best" way to start with this. In MeshCentral2InstallGuide.pdf there is a hint on page 23 to use --cert option, but this will create a private cert with this name? But this won't be accepted by default because it's private. So i will get a cert from my colleague. But how could i use MC with a "real" cert?

**Comment by si458** [comment]
Yes meshcentral allows you to use your own certificate I just can't remember how!
The docs also need updating to show how to use your own ssl cert and let's encrypt correctly! So I will look at updating those for you!

**Reply by simpsonetti** [reply]
Thanks a lot!

**Reply by si458** [reply]
The is already info in the docs that explain about ssl certs but doesn't really explain how to do it, so I will update docs when I get chance for you https://ylianst.github.io/MeshCentral/meshcentral/#server-certificate
But basically you remove the letsencrypt section and replace the certs in the data folder with your own and restart

**Reply by simpsonetti** [reply]
Okay so when i've got my webserver cert and key, where do i get root-cert, mpsserver and codesign and agentserver certs and key from?

**Reply by si458** [reply]
If a legitimate trusted certificate is available, replace “webserver-cert-public.crt” and “webserver-cert-private.key” with your certificate. These files are located in “meshcentral-data” folder of the server. The other certs can all be left self signed as they are used for different things like intel amt, windows exe signing and agents connecting

**Reply by simpsonetti** [reply]
Hm okay, but when i try to use my cert for windows.exe signing and agents connecting, what i have to do?

**Reply by si458** [reply]
Plz watch the video in the docs for codesigning https://ylianst.github.io/MeshCentral/meshcentral/codesigning/#automatic-agent-code-signing

**Reply by silversword411** [reply]
Why don't you have @si458 login to your server and set it up for you?

He can do that.
